### PR TITLE
45787 Set up health check

### PR DESCRIPTION
**Pull request by vipin14904**

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/UKHO.MaritimeSafetyInformation.Common.csproj [hunk 1]
```diff
@@ -6,9 +6,16 @@
...
 
+  <ItemGroup>
+    <None Include="Logging\EventIds.cs" />
+  </ItemGroup>
+
   <ItemGroup>
     <PackageReference Include="Azure.Messaging.EventHubs" Version="5.6.2" />
+    <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions" Version="6.0.4" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="6.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/UKHO.MaritimeSafetyInformation.Common.csproj:17]: Is "Newtonsoft.Json" package referred anywhere? Is it required?

**Review by usha13529 — APPROVED** [review]
LGTM.

**Review by usha13529 — APPROVED** [review]
LGTM.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/IEventHubLoggingHealthClient.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace UKHO.MaritimeSafetyInformation.Web.HealthCheck
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/IEventHubLoggingHealthClient.cs:3]:
```suggestion
namespace UKHO.MaritimeSafetyInformation.Common.HealthCheck
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthClient.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using UKHO.Logging.EventHubLogProvider;
+using UKHO.MaritimeSafetyInformation.Common.Configuration;
+using UKHO.MaritimeSafetyInformation.Common.Logging;
+
+namespace UKHO.MaritimeSafetyInformation.Web.HealthCheck
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthClient.cs:14]:
```suggestion
namespace UKHO.MaritimeSafetyInformation.Common.HealthCheck
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using UKHO.MaritimeSafetyInformation.Common.Logging;
+
+namespace UKHO.MaritimeSafetyInformation.Web.HealthCheck
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthCheck.cs:5]:
```suggestion
namespace UKHO.MaritimeSafetyInformation.Common.HealthCheck
```

**Review by barrie-cooper — COMMENTED** [review]
Just the namespaces need changing please

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common.UnitTests/UKHO.MaritimeSafetyInformation.Common.UnitTests.csproj [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common.UnitTests/UKHO.MaritimeSafetyInformation.Common.UnitTests.csproj:5]: Remove please
```suggestion
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/IEventHubLoggingHealthClient.cs (hunk 2 above)
Comment (vipin14904) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/IEventHubLoggingHealthClient.cs]: Resolved

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthClient.cs (hunk 3 above)
Comment (vipin14904) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthClient.cs]: Resolved

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common.UnitTests/UKHO.MaritimeSafetyInformation.Common.UnitTests.csproj (hunk 5 above)
Comment (vipin14904) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common.UnitTests/UKHO.MaritimeSafetyInformation.Common.UnitTests.csproj]: Removed

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthCheck.cs (hunk 4 above)
Comment (vipin14904) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/EventHubLoggingHealthCheck.cs]: Resolved

**Review by mattwin1 — APPROVED** [review]
Reviewed Devops-related files. LGTM 👍

**Review by barrie-cooper — APPROVED** [review]
LGTM, thanks @vipin14904

**Pull request merged**